diff --git a/ext/posix/posix.c b/ext/posix/posix.c
index e6655f0d..f3864855 100644
--- a/ext/posix/posix.c
+++ b/ext/posix/posix.c
@@ -710,11 +710,15 @@ int php_posix_group_to_array(struct group *g, zval *array_group) /* {{{ */
 	array_init(&array_members);
 
 	add_assoc_string(array_group, "name", g->gr_name);
+#ifdef __MVS__
+  add_assoc_null(array_group, "passwd");
+#else
 	if (g->gr_passwd) {
 		add_assoc_string(array_group, "passwd", g->gr_passwd);
 	} else {
 		add_assoc_null(array_group, "passwd");
 	}
+#endif
 	for (count = 0;; count++) {
 		/* gr_mem entries may be misaligned on macos. */
 		char *gr_mem;
@@ -902,10 +906,14 @@ int php_posix_passwd_to_array(struct passwd *pw, zval *return_value) /* {{{ */
 		return 0;
 
 	add_assoc_string(return_value, "name",      pw->pw_name);
+#ifndef __MVS__
 	add_assoc_string(return_value, "passwd",    pw->pw_passwd);
+#endif
 	add_assoc_long  (return_value, "uid",       pw->pw_uid);
-	add_assoc_long  (return_value, "gid",		pw->pw_gid);
+	add_assoc_long  (return_value, "gid",       pw->pw_gid);
+#ifndef __MVS__
 	add_assoc_string(return_value, "gecos",     pw->pw_gecos);
+#endif
 	add_assoc_string(return_value, "dir",       pw->pw_dir);
 	add_assoc_string(return_value, "shell",     pw->pw_shell);
 	return 1;
