diff --git a/ext/standard/syslog.c b/ext/standard/syslog.c
index 70fa8e24..f7e3ea75 100644
--- a/ext/standard/syslog.c
+++ b/ext/standard/syslog.c
@@ -50,7 +50,9 @@ PHP_MINIT_FUNCTION(syslog)
 	REGISTER_LONG_CONSTANT("LOG_MAIL", LOG_MAIL, CONST_CS | CONST_PERSISTENT); /* log to email */
 	REGISTER_LONG_CONSTANT("LOG_DAEMON", LOG_DAEMON, CONST_CS | CONST_PERSISTENT); /* other system daemons */
 	REGISTER_LONG_CONSTANT("LOG_AUTH", LOG_AUTH, CONST_CS | CONST_PERSISTENT);
+#ifndef __MVS__
 	REGISTER_LONG_CONSTANT("LOG_SYSLOG", LOG_SYSLOG, CONST_CS | CONST_PERSISTENT);
+#endif
 	REGISTER_LONG_CONSTANT("LOG_LPR", LOG_LPR, CONST_CS | CONST_PERSISTENT);
 #ifdef LOG_NEWS
 	/* No LOG_NEWS on HP-UX */
