diff --git a/Zend/zend_alloc.c b/Zend/zend_alloc.c
index edcf1192..63bd58ba 100644
--- a/Zend/zend_alloc.c
+++ b/Zend/zend_alloc.c
@@ -446,7 +446,7 @@ static void zend_mm_munmap(void *addr, size_t size)
 		}
 	}
 #else
-	if (munmap(addr, size) != 0) {
+	if (zend_munmap(addr, size) != 0) {
 #if ZEND_MM_ERROR
 		fprintf(stderr, "\nmunmap() failed: [%d] %s\n", errno, strerror(errno));
 #endif
@@ -480,7 +480,7 @@ static void *zend_mm_mmap_fixed(void *addr, size_t size)
 	flags |= MAP_TRYFIXED;
 #endif
 	/* MAP_FIXED leads to discarding of the old mapping, so it can't be used. */
-	void *ptr = mmap(addr, size, PROT_READ | PROT_WRITE, flags /*| MAP_POPULATE | MAP_HUGETLB*/, ZEND_MM_FD, 0);
+	void *ptr = zend_mmap(addr, size, PROT_READ | PROT_WRITE, flags /*| MAP_POPULATE | MAP_HUGETLB*/, ZEND_MM_FD, 0);
 
 	if (ptr == MAP_FAILED) {
 #if ZEND_MM_ERROR && !defined(MAP_EXCL) && !defined(MAP_TRYFIXED)
@@ -520,7 +520,7 @@ static void *zend_mm_mmap(size_t size)
 #else
 		fd = VM_FLAGS_SUPERPAGE_SIZE_2MB;
 #endif
-		ptr = mmap(NULL, size, PROT_READ | PROT_WRITE, mflags, fd, 0);
+		ptr = zend_mmap(NULL, size, PROT_READ | PROT_WRITE, mflags, fd, 0);
 		if (ptr != MAP_FAILED) {
 			zend_mmap_set_name(ptr, size, "zend_alloc");
 			return ptr;
@@ -528,7 +528,7 @@ static void *zend_mm_mmap(size_t size)
 	}
 #endif
 
-	ptr = mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, ZEND_MM_FD, 0);
+	ptr = zend_mmap(NULL, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, ZEND_MM_FD, 0);
 
 	if (ptr == MAP_FAILED) {
 #if ZEND_MM_ERROR
