diff --git a/Zend/zend_fibers.c b/Zend/zend_fibers.c
index 6a22d5c4..1cb8dc48 100644
--- a/Zend/zend_fibers.c
+++ b/Zend/zend_fibers.c
@@ -211,13 +211,19 @@ static zend_fiber_stack *zend_fiber_stack_allocate(size_t size)
 	}
 # endif
 #else
+# ifdef __MVS__
+	pointer = calloc(alloc_size, 1);
+	if (!pointer) {
+		zend_throw_exception_ex(NULL, 0, "Fiber stack allocate failed: mmap failed: %s (%d)", strerror(errno), errno);
+		return NULL;
+	}
+# else
 	pointer = mmap(NULL, alloc_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_STACK, -1, 0);
 
 	if (pointer == MAP_FAILED) {
 		zend_throw_exception_ex(NULL, 0, "Fiber stack allocate failed: mmap failed: %s (%d)", strerror(errno), errno);
 		return NULL;
 	}
-
 	zend_mmap_set_name(pointer, alloc_size, "zend_fiber_stack");
 
 # if ZEND_FIBER_GUARD_PAGES
@@ -227,6 +233,7 @@ static zend_fiber_stack *zend_fiber_stack_allocate(size_t size)
 		return NULL;
 	}
 # endif
+#endif
 #endif
 
 	zend_fiber_stack *stack = emalloc(sizeof(zend_fiber_stack));
